File: problems/leetcode/daily-questions/DCP-06-2024/80-Remove-Duplicates-from-Sorted-Array-II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
    int prev=0,curr=1,n=nums.size();
    while(curr<n){
        if(nums[prev]==nums[curr]){
            if(curr-prev!=1){
                nums[curr]= 10001;
            }
        }
        else 
            prev=curr;
        curr++;
    }
    sort(nums.begin(),nums.end());
    auto pos=lower_bound(nums.begin(),nums.end(), 10001);
    nums.erase(pos,nums.end());
    return nums.size();
    }
};

File: problems/leetcode/150-interview/80-Remove-Duplicates-from-Sorted-Array-II.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        if (nums.Length <= 2) return nums.Length;
        int idx = 2; 
        for (int i = 2; i < nums.Length; i++) {
            if (nums[i] != nums[idx - 2]) {
                nums[idx++] = nums[i]; 
            }
        }
        return idx; 
    }
}